include/circt/Dialect/ESI/ESIChannels.td

```diff
@@ -197,6 +197,78 @@ def UnwrapSVInterfaceOp : ESI_Op<"unwrap.iface", [
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Channel bundles
+//===----------------------------------------------------------------------===//
+
+// Used to indicate the presence of a keyword in asm. Should be used within
+// optional parens in the declarative assembly format.
+// Upstreaming: https://github.com/llvm/llvm-project/pull/65438
+def UnitParameter : AttrOrTypeParameter<"::mlir::UnitAttr", "boolean flag"> {
+  let printer = "";
+  let parser = "::mlir::UnitAttr::get($_parser.getContext())";
+  let defaultValue = "::mlir::UnitAttr()";
+}
+
+def ChannelDirection : I32EnumAttr<"ChannelDirection",
+  "Direction of channel", [
+    I32EnumAttrCase<"to", 1>,
+    I32EnumAttrCase<"from", 2>,
```

Comment:
Not sure i'm convinced that to/from is the right terminology here, although i understand the reasoning. Choosing in/out would i suspect be a bit easier to understand, since it's simpler (for me) to predict what values i need to provide for a pack/unpack operation of this bundle type.

Reply:
See, I think in/out is more confusing. I'm not convinced there is any good terminology here. The other problem with in/out is those words are pretty heavily overloaded.